docs/core/heading.md

```diff
@@ -0,0 +1,76 @@
+# Heading
+
+The flexible and hybrid modes allow adding a desired heading (north based azimuth between 0 and 360 degree) to any point. Adding a heading makes it more likely that a route starts towards the provided direction, because roads going into other directions are penalized (see the [heading_penalty](#heading-penalty) parameter).
+
+A heading with the value 'NaN' won't be enforced and a heading not within [0, 360] will trigger an IllegalStateException. It is important to note that if you force the heading at via or end points the outgoing heading needs to be specified. I.e. if you want to force "coming from south" to a destination you need to specify the resulting "heading towards north" instead, which is 0.
+
+## Example
+
+For this example we use [andorra.osm.pbf](../../core/files/andorra.osm.pbf).
+
+You can follow this example in 3 ways:
+
+1.	Pure Java without server ([HeadingExample.java](../../example/src/test/java/com/graphhopper/example/HeadingExample.java))
+2.	With Java client-hc requests to server ([HeadingExamples.java](../../client-hc/src/test/java/com/graphhopper/api/HeadingExamples.java))
+3.	With common get request to server (the urls used in the following)
+
+For 2 and 3 you can self-host a graphhopper web server as described [here](https://github.com/graphhopper/graphhopper#installation), but in [config-example.yml](../../config-example.yml) we need to outcomment
+
+```
+  profiles_ch:
+    - profile: car
+```
+
+in line 56-57 since `heading` is not available in speed mode.
+
+The docs for the API can be found [here](../../docs/web/api-doc.md#parameters)
+
+### Use Case
+
+We will route from point
+`42.566757, 1.597751`
+to point
+`42.567396, 1.597807`
+
+#### Without Heading
+
+`localhost:8989/route?profile=car&point=42.566757,1.597751&point=42.567396,1.597807&type=json&instructions=false&points_encoded=false`
+
+![without_heading](./images/without_heading.PNG)
+
+(the images are created by putting the resulting coordinates LineString in a GeoJSON and plotting them)

```

Comment:
Unfortunately we cannot set the heading in GraphHopper Maps yet: https://github.com/graphhopper/graphhopper-maps/issues/297 Then we could simply create screenshots with it :)

Reply:
at the moment we are also thinking about a good way to integrate heading in our UI (optional parameter for each point) :)

Reply:
In the previous version of GraphHopper Maps things like this were possible by manually modifying the url, but that isn't ideal either.